### ISSUE TITLE
"isVisible" is incorrectly assessed for the absolutely positioned elements if the ancestor has overflow and static position

### ISSUE THREAD

**Issue by klarzynskik** [cypress-io/cypress#28638]
### Current behavior

It appears "isVisible" is incorrectly assessed for the absolutely positioned elements if the ancestor has an overflow and static position. For example, for the following document the `#visible-button` is considered as not-visible:

```html
<body>
  <div style="height: 200px; position: relative; display: flex">
    <div style="border: 5px solid red">
      <div
        id="breaking-container"
        style="overflow: auto; border: 5px solid green"
      >
        <div>
          <h1>Example</h1>
          <div style="position: absolute; bottom: 5px">
            <button id="visible-button" style="position: relative">
              Try me
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
```

Here's the screenshot from the test:

![button shown as not visible despite being there](https://github.com/cypress-io/cypress/assets/47242672/ff85c2a5-bf35-4002-ac8b-ffb178d5dd5e)

It appears that in this case, while assessing the `isVisible`, the button goes particularly into the [isHidddenByAncestors](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/dom/visibility.ts#L101). Then while calculating the [elIsOutOfBoundsOfAncestorsOverflow](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/dom/visibility.ts#L311) for the `#breaking-container` div element from the example, the `canClipContent` is incorrectly assessed. That element has a default `position: static`, thus does not affect the visibility of the button. However, this is not checked in the `canClipContent`, resulting in the incorrectly assessed visibility via calculating the button's position compared to that div.

The [documentation](https://docs.cypress.io/guides/core-concepts/interacting-with-elements#An-element-is-considered-hidden-if) in that particular states that:

> Any of its ancestors hides overflow*
    AND the element is position: relative
    AND it is positioned outside that ancestor's bounds
    
 However, as mentioned above, I think that this should be true only if the considered ancestor's position is not static.

### Desired behavior

Element visibility is correctly assessed.

### Test code to reproduce

Here's the repository: https://github.com/klarzynskik/cypress-visibility-issue-demo

The simplest example is the following code:

```html
<div id="breaking-container" style="overflow: auto">
      <div>
        <div style="position: absolute; bottom: 5px">
          <button id="visible-button">Try me</button>
        </div>
      </div>
</div>
```

### Cypress Version

13.6.2

### Node version

18.16.1

### Operating System

macOS 13.4.1

**Comment by jennifer-shehane** [comment]
@klarzynskik We would be open to accepting a PR to update our visibility algorithm to address this. The code for visibility is here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/dom/visibility.ts

**Comment by klarzynskik** [comment]
@jennifer-shehane, thanks for the reply! Indeed, I was already looking into that, and it seems that the most straightforward option would be to move to the native `Element.checkVisibility` (chromium feature description: https://chromestatus.com/feature/5163102852087808), which has been widely adopted in other frameworks (i.e. Playwright) and is supported in all modern browsers.

Would you be open to such a refactor? I would be happy to work on that.

**Comment by jennifer-shehane** [comment]
@klarzynskik Yes! I had spent about 20 minutes trying to throw this check in place of our visibility checks one day and some of our tests were failing, but I didn't have time to investigate further. 

As long as our current visibility tests all pass - any updates and refactors are welcome, especially if they address any of the [outstanding visibility issues](https://github.com/cypress-io/cypress/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+visibility+%F0%9F%91%81%22) with added tests and do not impact performance negatively.

**Comment by senpl** [comment]
I checked in my version and seems like:
https://github.com/cypress-io/cypress/pull/29224
fix also this issue.

**Comment by jennifer-shehane** [comment]
@senpl Thanks, I'll note that in the PR for us to verify also.

**Comment by senpl** [comment]
> Would you be open to such a refactor? I would be happy to work on that.

I already did that refactor in https://github.com/cypress-io/cypress/pull/29741 . But your comment make me aware of this solution.
